### PR TITLE
feat: Added checks workflow

### LINKED ISSUE #235: Prettifying markdown files
My proposal to standardize markdown files is using [prettier](https://prettier.io/). It uses default prettier config + additional wrap option `"proseWrap": "always"`.

**Why do we need this?**
Documentation about standard should be standardified! =)

**How it works?**
We are not using JS / NPM / package.json, so I can't write some script to make it automatically, but I can leave config. Each contributor can make it manually:
[VSCode+Prettier](https://www.loom.com/share/326db3432ea040ce98027efffad598ae)

**Next steps:**
If someone confident with git hooks and prettifying for Ruby welcome to make it automatically.

I can implement it with JS stack, but does it'll be okay to mix stacks?

here is PR: #234

### PR DESCRIPTION
**Pull request by alexandrtovmach**
Workflow to validate changes and protect spec content from direct updates. Added linter for markdown and standardized contentbase.

## Why it is?

1. We should validate changes by automated tools
2. SemVer project contain two repositories, one for spec, another one for website. This PR should solve problem with direct spec updates not in a right place. All changes about SemVer should be in **semver/semver**, not here, so this workflow will warn users with message:

message

## How it works?

In case that GitHub Actions doesn't provide access to `GITHUB_TOKEN` which required for writing comment to PR, we should trigger it manually. Its normal practice at the current moment, and to trigger checks need print one comment to PR with `/check`:

image

### CODE REVIEW

**Review by indirect — CHANGES_REQUESTED** [review]
This changes the rendered version of index.md, with not just indentation but also content. I don't think we want an automatic formatter to change the content of the SemVer spec!

You can see the rendered diff, and the indent changes to BNF and content changes to regex, here: https://github.com/semver/semver.org/pull/266/files?short_path=d680e8a#diff-d680e8a854a7cbad6d490c445cba2eba

**Comment by alexandrtovmach** [comment]
@indirect Yeah, it's my mistake with `regex`, not formatter :disappointed: I did global "Find & Replace" and did in wrong way. Thanks for pay attention :+1: 

Investigated a bit more I figured out with config to avoid drastic changes. Formatting changes:
- unordered-list-marker-style  to use `"-"` (see example https://github.com/semver/semver.org/pull/266/files#diff-fe45e64f3f9908cf45fb26a75a3a3a15L277-L285)
- code-block-style to use `"fenced"`, with fixed indentation :wink:  (see example https://github.com/semver/semver.org/pull/266/files#diff-963215f81f79799fe04574157522069dL153-R216)
- removed no-consecutive-blank-lines (see example https://github.com/semver/semver.org/pull/266/files#diff-d680e8a854a7cbad6d490c445cba2ebaL235)
- mdash-style to use `"―"` (see example https://github.com/semver/semver.org/pull/266/files#diff-2f728cf67420f4f4c507ccfe1cd7320dL16-L18)
- removed consecutive spaces (see example https://github.com/semver/semver.org/pull/266/files#diff-2f728cf67420f4f4c507ccfe1cd7320dL16-L18)

**Comment by alexandrtovmach** [comment]
@indirect could you, if this PR looks okay, approve it, merge and then turn on rule to require passing workflow check for PRs to gh-pages branch? I haven't enough permissions to do that :shrug:, so that's can be done by you or just grant me admin rights if it's not a problem

**Comment by alexandrtovmach** [comment]
NOTE: Our default branch is `gh-pages`, so please ignore `master` on screenshots

Settings -> Branches -> Edit rule (if exist) or Add rule:
image

Set "Require status checks to pass before merging" and "Require branches to be up to date before merging" to true and select our check workflow from list (should available after merge)
image

**Review by indirect — COMMENTED** [review]
@alexandrtovmach looks like there's a few places in the BNF fenced code block where the rendered output is indented differently now. Can you update index.md so the rendered versions stays the same?

After that, I'm ready to merge this and turn on the setting. 👍

**Comment by alexandrtovmach** [comment]
@indirect Done, please take a look

**Review by indirect — APPROVED** [review]

**Comment by indirect** [comment]
hmm I tried to turn on the setting, but this repo has not yet seen the "check" status 😅 maybe after someone runs it?

**Comment by alexandrtovmach** [comment]
@indirect it's available after merge

**Comment by indirect** [comment]
Done!

image

**Comment by alexandrtovmach** [comment]
@indirect -__- something wrong, it's not report the status to PR, will check

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@0.9.0
+        id: comment
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: return context.payload.issue.pull_request && context.payload.comment.body === '/check'
```
Comment (marilouzamora408-ai) [inline .github/workflows/checks.yml:15]: Thanks for su
Screenshot_2025-10-04-01-35-17-99
pport

**Pull request merged**